From 78107cb3e347550567a0cec38f4baa806f60828d Mon Sep 17 00:00:00 2001
From: Philipp Zabel <philipp.zabel@gmail.com>
Date: Fri, 6 Mar 2020 00:03:49 +0100
Subject: [PATCH 39/45] glupload: dmabuf: only accept uploads to external-oes
 if supported by the context

This keeps the DirectDmabufExternal uploader from accepting buffers if
texture-target=external-oes is not supported by the GL context.

(cherry picked from commit 40773e9c91651300aefd4e89da8766d7d05d23d7)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 gst-libs/gst/gl/gstglupload.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gst-libs/gst/gl/gstglupload.c b/gst-libs/gst/gl/gstglupload.c
index 730a073b99..6ee1195ec9 100644
--- a/gst-libs/gst/gl/gstglupload.c
+++ b/gst-libs/gst/gl/gstglupload.c
@@ -635,6 +635,11 @@ _dma_buf_upload_accept (gpointer impl, GstBuffer * buffer, GstCaps * in_caps,
           "EGL_KHR_image_base"))
     return FALSE;
 
+  if (dmabuf->target == GST_GL_TEXTURE_TARGET_EXTERNAL_OES &&
+      !gst_gl_context_check_feature (dmabuf->upload->context,
+          "GL_OES_EGL_image_external"))
+    return FALSE;
+
   /* This will eliminate most non-dmabuf out there */
   if (!gst_is_dmabuf_memory (gst_buffer_peek_memory (buffer, 0)))
     return FALSE;
-- 
2.11.0

